### PR TITLE
Don't send network messages until network is ready

### LINKED ISSUE #759: If node is not synchronized, it should neither send consensus messages nor send monetary blocks.
If node have outdated blockchain history, it should not flood network with consensus, or block messages.

### PR DESCRIPTION
**Pull request by rtsisyk**
Fix a race on start - don't send the key && micro blocks and
the view changes until network is ready.

Closes #759

### CODE REVIEW

File: network/src/gatekeeper/behavior.rs [hunk 1]
```diff
@@ -408,7 +408,9 @@ where
                     self.solvers.insert(peer_id.clone());
                     thread::spawn(move || {
                         let start = SystemTime::now();
+                        info!("Solving a hashcash puzzle: peer_id={:?}", peer_id);
                         let proof = hashcash::delay(p.nbits, &p.seed);
+                        info!("Solved a hashcash puzzle: peer_id={:?}", peer_id);
```
Comment (vldm) [inline network/src/gatekeeper/behavior.rs]: may be add start.elapsed to output?

Comment (echupriyanov) [reply network/src/gatekeeper/behavior.rs]: Can you create a separate issue for this?

Comment (vldm) [reply network/src/gatekeeper/behavior.rs]: Why we need separete issue for "one-line" fixes, that is also related to pull request changes?

Comment (echupriyanov) [reply network/src/gatekeeper/behavior.rs]: Network layer fixes doesn't belong here.

**Pull request merged**